src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -327,26 +328,43 @@ public void moveElementsDirectory(List<UUID> elementsUuids, UUID newDirectoryUui
 
         validateNewDirectory(newDirectoryUuid);
 
-        elementsUuids.forEach(elementUuid -> moveElementDirectory(elementUuid, newDirectoryUuid, userId));
+        elementsUuids.forEach(elementUuid -> moveElementDirectory(getDirectoryElementEntity(elementUuid), newDirectoryUuid, userId));
     }
 
-    private void moveElementDirectory(UUID elementUuid, UUID newDirectoryUuid, String userId) {
-        DirectoryElementEntity element = repositoryService.getElementEntity(elementUuid)
-                .orElseThrow(() -> DirectoryException.createElementNotFound(ELEMENT, elementUuid));
+    private void moveElementDirectory(DirectoryElementEntity element, UUID newDirectoryUuid, String userId) {
+        if (element.getParentId() == newDirectoryUuid) { // Same directory ?
+            return;
+        }
```

Comment:
small bug here , '==' does not work on uuids
```suggestion
        if (element.getParentId().equals(newDirectoryUuid)) { // Same directory ?
            return;
        }
```

may be we need a small unit test for this case also ?